src/Element.cc

```diff
@@ -222,24 +228,19 @@ void Element::Copy(const ElementPtr _elem)
   for (Param_V::iterator iter = _elem->dataPtr->attributes.begin();
        iter != _elem->dataPtr->attributes.end(); ++iter)
   {
-    if (!this->HasAttribute((*iter)->GetKey()))
+    if (this->HasAttribute((*iter)->GetKey()))
     {
-      this->dataPtr->attributes.push_back((*iter)->Clone());
+      this->RemoveAttribute((*iter)->GetKey());
     }
-    ParamPtr param = this->GetAttribute((*iter)->GetKey());
-    (*param) = (**iter);
+    auto clonedAttribute = (*iter)->Clone();
+    clonedAttribute->SetParentElement(shared_from_this());
+    this->dataPtr->attributes.push_back(clonedAttribute);
```

Comment:
Is there a reason for using `sdf::Param::Clone` instead of the original `sdf::Param::operator=` here? I think removing and adding is more expensive than assigning the value for an existing attribute.

Reply:
Thanks for catching this, this was modified some time during development as I was figuring out whether cloning and copying should handle parent element allocation. https://github.com/ignitionrobotics/sdformat/pull/589/commits/03bf815f5610d2f3ac44cf54b7a51f042b9d5b38